src/main/deltachat/webxdc.ts

```diff
@@ -248,97 +249,189 @@ export default class DCWebxdc extends SplitOut {
         internet_access: webxdcInfo['internetAccess'],
       }
 
+      type permission_arg = Parameters<
+        Exclude<
+          Parameters<
+            | typeof webxdcWindow.webContents.session.setPermissionCheckHandler
+            | typeof webxdcWindow.webContents.session.setPermissionRequestHandler
+          >[0],
+          null
+        >
+      >[1]
+      // TODO some (poorly written?) apps might require a refresh
+      // after a permission has been granted,
+      // but we don't support it because
+      // 1. There is no way to refresh a webxdc
```

Comment:
app can call `window.location.reload()` in it's iframe, you can only not reload the parent page.